### PR TITLE
feat: RecordContainerJsonMarshaller tool to marshall record container…

### PR DESCRIPTION
**Pull request by bartlomiejmarszal**
In order to store data in RDS without losing data we would like to store it in a more controlled way. 

Related to https://github.com/oat-sa/extension-tao-outcomeui/pull/446

### CODE REVIEW

**Comment by julien-sebire** [comment]
This looks good to me. Good thing to have put it in this place.
I'm not able to test it now, though.

File: helpers/class.ResultTransmitter.php [hunk 1]
```diff
@@ -182,6 +185,11 @@ private static function transformValue($value) {
             if ($value instanceof QtiFile) {
                 return taoQtiCommon_helpers_Utils::qtiFileToString($value);
             }
+            if ($value instanceof RecordContainer) {
+                $marshaller = new Marshaller;
```
Comment (andreluizmachado) [inline helpers/class.ResultTransmitter.php:189]: PSR12:  Parentheses must be used when instantiating a new class

```suggestion
                $marshaller = new Marshaller();
```

File: helpers/class.ResultTransmitter.php [hunk 2]
```diff
@@ -182,6 +185,11 @@ private static function transformValue($value) {
...
+                return $marshaller->marshall($value);
+            }
+
```
Comment (andreluizmachado) [inline helpers/class.ResultTransmitter.php:192]: PSR12: Expected 1 space after closing brace; newline found

Comment (andreluizmachado) [reply helpers/class.ResultTransmitter.php:192]: I'd recommend to check the file to see if there are more issues like this...

Comment (andreluizmachado) [reply helpers/class.ResultTransmitter.php:192]: would you mind to create a ticket for enabling the ci as we have for the extensions on this library?

Comment (bartlomiejmarszal) [reply helpers/class.ResultTransmitter.php:192]: Yes. I tried to include missing GitHub files but it seems not working for this lib.

**Review by pnal — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by andreluizmachado — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Pull request merged**